xmppserver/src/main/java/org/jivesoftware/openfire/muc/MUCRoom.java

```diff
@@ -2452,10 +2452,9 @@ public void sendInvitation(JID to, String reason, MUCRole senderRole, List<Eleme
                 }
             }
             Element frag = message.addChildElement("x", "http://jabber.org/protocol/muc#user");
-            // ChatUser will be null if the room itself (i.e. via admin console) made the request
-            if (senderRole.getUserAddress() != null) {
-                frag.addElement("invite").addAttribute("from", senderRole.getUserAddress().toBareJID());
-            }
+            // ChatUser will be null if the room itself (i.e. via admin console) made the request. In that case, use the room JID. See OF-2486.
+            final JID from = senderRole.getUserAddress() != null ? senderRole.getUserAddress() : getJID();
```

Comment:
Where the sender is identifiable, this changes it from bare to full MUC JID. Intentional?

Reply:
One line lower, there's a `asBareJID()` call for the `from` value.

Reply:
:facepalm: